### ISSUE TITLE
tensordot/conj interplay

### ISSUE THREAD

**Issue by mrader1248** [tensorflow/tensorflow#19771]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: CentOS Linux 7, Kernel 3.10
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: v1.8.0-0-g93bc2e2072 1.8.0
- **Python version**: Python 3.6.2 :: Continuum Analytics, Inc.
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: CUDA 9.0.176 (but not in use)
- **GPU model and memory**: GeForce GTX 1080 (but not in use)
- **Exact command to reproduce**: execute python script below

### Describe the problem
I am using the code below to contract four tensors `A[l,m,n] B[q,l,i] C[j,p,m,q] conj(D[p,n,k])` using a series of tensordot operations. To verify the result, I compare tensorflow with pure numpy. It seems like the `tf.conj` node is ignored (or an additional conjugation is executed) when executing tensorflow's `tensordot`s, as the first `print` gives me `False` and the second one `True`, although it should be the other way round. Just as a remark: The third part of my code uses `einsum` instead and works as intended (output of the third `print` should be `True` and is indeed `True`). Thank you for your help!

### Source code / logs
```python
import numpy as np
import tensorflow as tf

n = 100
m = 4
dtype = np.complex128

### generate random tensors ###

def rand(*shape):
    return np.asarray(
        np.random.random(shape) + 1j*np.random.random(shape),
        dtype
    )

A = rand(n, m, n)
B = rand(m, n, n)
C = rand(m, m, m, m)
D = rand(m, n, n)

### perform numpy computation ###

X_np = np.tensordot(A, B, (0,1))
Y_np = np.tensordot(X_np, C, ((0,2),(2,3)))
Z_np = np.tensordot(Y_np, D.conj(), ((3,0),(0,1)))

### build tensorflow computation graph ###

A_node = tf.placeholder(dtype, (n, m, n))
B_node = tf.placeholder(dtype, (m, n, n))
C_node = tf.placeholder(dtype, (m, m, m, m))
D_node = tf.placeholder(dtype, (m, n, n))

X_node = tf.tensordot(A_node, B_node, (0,1))
Y_node = tf.tensordot(X_node, C_node, ((0,2),(2,3)))
Z_node = tf.tensordot(Y_node, tf.conj(D_node), ((3,0),(0,1)))

### run tensorflow computation ###

session = tf.Session()
Z_tf = session.run(
    Z_node, feed_dict={A_node: A, B_node: B, C_node: C, D_node: D}
)
session.close()

### compare results ###

print(np.allclose(Z_np, Z_tf))

### run tensorflow computation again with numpy conj ###

session = tf.Session()
Z_tf = session.run(
    Z_node, feed_dict={A_node: A, B_node: B, C_node: C, D_node: D.conj()}
)
session.close()

### compare results ###

print(np.allclose(Z_np, Z_tf))

### build tensorflow computation graph using einsum ###

W_node = tf.einsum(
    "lmn,qli,jpmq,pnk->ijk", A_node, B_node, C_node, tf.conj(D_node)
)

session = tf.Session()
W_tf = session.run(
    W_node, feed_dict={A_node: A, B_node: B, C_node: C, D_node: D}
)
session.close()

print(np.allclose(Z_np, W_tf))
```
### Output
```
False
True
True
```

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Bazel version

**Comment by mrader1248** [comment]
Sorry, I deleted the fields Bazel version and GCC/Compiler version, as I was not installing tensorflow from source.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 89 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 104 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 119 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 134 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 149 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 167 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 181 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by mohantym** [comment]
Hi @mrader1248 ! This issue is getting resolved in 2.8 version using compat.v1 mode. Attaching [gist](https://colab.sandbox.google.com/gist/mohantym/af6bfffddad9426b1652a27091d9376e/git_19771.ipynb#scrollTo=jgg-kJ6h1fvB) for reference . Thanks!
